### PR TITLE
Update Actions and staticcheck, build with Go 1.20

### PR DESCRIPTION
**Pull request by bartekpacia**
- go.mod: bump Go to 1.20
- go-publish.yml: update and clean up Actions
- go-test.yml: update and clean up Actions

### CODE REVIEW

**Comment by bartekpacia** [comment]
Let's wait with merging until https://github.com/Homebrew/homebrew-core/pull/122082 is merged. Otherwise we won't be able to build it locally with `brew install`-ed version of Go.

File: .github/workflows/go-test.yml [hunk 1]
```diff
@@ -37,7 +35,7 @@ jobs:
       - name: Run linter
         if: always()
         run: |
-          go install honnef.co/go/tools/cmd/staticcheck@v0.3
+          go install honnef.co/go/tools/cmd/staticcheck@v0.4.0
```
Comment (Albert221) [inline .github/workflows/go-test.yml:38]: `0.4` means `0.4.x`. `0.4.0` is pinned to 0 as minor afaik

File: .github/workflows/go-test.yml [hunk 2]
```diff
@@ -51,32 +49,27 @@ jobs:
...
-          restore-keys: |
-            ${{ runner.os }}-go-
+          go-version: ${{ matrix.go-version }}
+          cache: true
 
       - name: Build binary
-        run: go install .
+        run: |
+          go build
+          mv ./poe2arb /usr/local/bin
```
Comment (Albert221) [inline .github/workflows/go-test.yml:67]: Why did you change it like that?

File: .github/workflows/go-test.yml (hunk 1 above)
Comment (bartekpacia) [reply .github/workflows/go-test.yml]: Changed to `0.4`

File: .github/workflows/go-test.yml (hunk 2 above)
Comment (bartekpacia) [reply .github/workflows/go-test.yml:67]: Because `go install .` doesn't work in Go 1.20 – the tests were failing with `poe2arb not found`.

I think it might be related to [this change](https://tip.golang.org/doc/go1.20#go-command):

> The directory $GOROOT/pkg no longer stores pre-compiled package archives for the standard library: go install no longer writes them, the go build no longer checks for them, and the Go distribution no longer ships them.

Comment (Albert221) [reply .github/workflows/go-test.yml:67]: No, it's not that. It must be something else. We'll figure it

**Comment by Albert221** [comment]
I wonder whether bumping the Go version is a good practice when we don't benefit from it directly (Go 1.19 added generics we wanted to clean up the code).

It widens the number of users that can `go install` the tool.

File: .github/workflows/go-test.yml [hunk 3]
```diff
@@ -8,26 +8,24 @@ on:
 
 jobs:
   test:
-    name: Go 1.19
+    name: Go ${{ matrix.go-version }}
+
+    strategy:
+      fail-fast: false
+      matrix:
```
Comment (Albert221) [inline .github/workflows/go-test.yml:15]: I'm not sure whether supporting different Go versions for our tool is something we need. It's a _tool_, ideally used only as a binary. Compiling it from source is one of the ways to install it, but I wouldn't call it something we officially support

Comment (bartekpacia) [reply .github/workflows/go-test.yml:15]: > I'm not sure whether supporting different Go versions for our tool is something we need.

Strong agree.

> Compiling it from source is one of the ways to install it, but I wouldn't call it something we officially support

Strong agree as well.

Let's use the latest stable Go release :)

File: .github/workflows/go-publish.yml [hunk 4]
```diff
@@ -16,10 +16,10 @@ jobs:
         with:
           fetch-depth: 0
 
-      - name: Setup Go
+      - name: Set up Go
         uses: actions/setup-go@v3
         with:
-          go-version: '1.20'
+          go-version-file: go.mod
```
Comment (Albert221) [inline .github/workflows/go-publish.yml]: :O

File: .github/workflows/go-test.yml [hunk 5]
```diff
@@ -51,27 +44,22 @@ jobs:
...
-          go-version: '1.19'
-
-      - name: Cache Go modules
-        uses: actions/cache@v2
-        with:
-          path: ~/go/pkg/mod
-          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
-          restore-keys: |
-            ${{ runner.os }}-go-
+          go-version: ${{ matrix.go-version }}
```
Comment (Albert221) [inline .github/workflows/go-test.yml:56]: There's no `go-version` in matrix for this job

File: .github/workflows/go-test.yml (hunk 2 above)
Comment (bartekpacia) [reply .github/workflows/go-test.yml:62]: i'm gonna ask a fellow gopher

@shilangyu any ideas?
